packages/server/src/api/controllers/datasource.ts

```diff
@@ -183,9 +183,7 @@ export async function buildSchemaFromDb(ctx: UserCtx) {
 
   let { tables, error } = await buildSchemaHelper(datasource)
   if (tablesFilter) {
-    if (!datasource.entities) {
```

Comment:
How come will this not be populated anymore?

Reply:
This was actually causing a bug with the old table filters, not resetting the entities structure before each filter meant that you couldn't remove already added tables - not super important given this is all being updated shortly anyway.